django/db/models/base.py

```diff
@@ -1923,6 +1924,30 @@ def _check_constraints(cls, databases):
                         id='models.W038',
                     )
                 )
+            def fetch_check_constraint_check_fields(check):
+                fields = set()
+                if isinstance(check, (ExpressionWrapper, RawSQL)):
+                    return fields
```

Comment:
I don't know how to extract `fields` from `CheckConstraint` with `ExpressionWrapper` and  `RawSQL` check.
Any idea?

Reply:
I this is not possible for `RawSQL` as these objects are opaque by design. `ExpressionWrapper` you should be able to use `get_source_expressions`?

Reply:
Thanks @charettes. Yes `get_source_expressions` works for `ExpressionWrapper`